tests/testsuite/vendor.rs

```diff
@@ -596,6 +596,62 @@ fn git_simple() {
     assert!(csum.contains("\"package\":null"));
 }
 
+#[cargo_test]
+fn git_diff_rev() {
+    let (git_project, git_repo) = git::new_repo("git", |p| {
+        p.file("Cargo.toml", &basic_manifest("a", "0.1.0"))
+            .file("src/lib.rs", "")
+    });
+    let url = git_project.url();
+    let ref_1 = "v0.1.0";
+    let ref_2 = "v0.2.0";
+
+    git::tag(&git_repo, ref_1);
+
+    git_project.change_file("Cargo.toml", &basic_manifest("a", "0.2.0"));
+    git::add(&git_repo);
+    git::commit(&git_repo);
+    git::tag(&git_repo, ref_2);
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            &format!(
+                r#"
+                    [package]
+                    name = "foo"
+                    version = "0.1.0"
+
+                    [dependencies]
+                    a_1 = {{ package = "a", git = '{url}', rev = '{ref_1}' }}
+                    a_2 = {{ package = "a", git = '{url}', rev = '{ref_2}' }}
+                "#
+            ),
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("vendor --respect-source-config")
+        .stream()
+        .with_stdout(
+            r#"
+[source.[..]]
```

Comment:
I believe since #11273 that the newline is no longer included. 

Also, I'd like to see the actual format in the test here instead of using `[..]`.

```suggestion
            r#"[source."git+file://[..]/git?rev=v0.1.0"]
```